### PR TITLE
Add timestamp timezone orc tests

### PR DESCRIPTION
**Pull request by yimngli-vmware**
Generate ORC files from JSON files instead of CSV file to handle nested array in the test data. Add more tests in the Makefile. Add TIMESTAMP WITH TIME ZONE into ORC resolver unit tests.

### CODE REVIEW

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java [hunk 1]
```diff
@@ -356,6 +356,25 @@ public static OneField[] timestampMapper(VectorizedRowBatch batch, ColumnVector
         return result;
     }
 
+    public static OneField[] timestampWithTimeZoneMapper(VectorizedRowBatch batch, ColumnVector columnVector, Integer oid) {
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java:359]: This method looks very similar to `timestampMapper`, with the only exception being the call to the helper method. I wonder if we can refactor this to avoid the duplication.

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedBaseTest.java [hunk 2]
```diff
@@ -30,16 +31,17 @@ public class ORCVectorizedBaseTest {
...
-    static final Boolean[] COL9_SUBSET = {false, true, false, true, false, true, false, true, false, true, false, false, false, false, false, null, false};
-    static final Short[] COL11_SUBSET = {10, 20, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1100, 1100, 1100, 1100, 1100, 1100};
-    static final String[] COL13_SUBSET = {"abcd", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", "abcde", null};
+    static final String[] COL7_SUBSET ={"2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07","2020-06-28 11:30:00-07"};
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedBaseTest.java:41]: Same comment as above regarding varying the timestamp values.

File: server/pxf-hdfs/src/test/resources/orc/Makefile [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# default assumes ORC tools JAR is downloaded in current directory
+# download from https://search.maven.org/artifact/org.apache.orc/orc-tools/1.7.5/jar
+ORC_TOOLS_JAR ?= orc-tools-1.7.5-uber.jar
+
+# the CSV data files use '\N' for null columns but the CSV parsing in orc-tools discards '\'
+# therefore the proper null string for orc-tools convert is 'N'
+CSV_NULL = N
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/resources/orc/Makefile:7]: I don't think this variable is being used anymore since we switched to JSON.

File: server/pxf-hdfs/src/test/resources/orc/README.md [hunk 4]
```diff
@@ -9,30 +9,23 @@ follow the instructions below.
 - Hadoop CLI commands
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/resources/orc/README.md:9]: Since we are using the ORC tools JAR now, I don't think we need Hadoop or Hive.

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java [hunk 5]
```diff
@@ -24,7 +24,9 @@
 import java.time.Instant;
 import java.time.LocalDate;
 import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
 import java.util.Arrays;
+import java.util.function.Function;
```
Comment (denalex) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java:29]: The PR needs to be against the most current ORC-write branch `orc-write-tests`, not master, as that branch has support for reading that type. There should ideally be no changes to this class here.

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java [hunk 6]
```diff
@@ -20,30 +20,7 @@
...
-import static org.greenplum.pxf.api.io.DataType.NUMERIC;
-import static org.greenplum.pxf.api.io.DataType.REAL;
-import static org.greenplum.pxf.api.io.DataType.SMALLINT;
-import static org.greenplum.pxf.api.io.DataType.TEXT;
-import static org.greenplum.pxf.api.io.DataType.TEXTARRAY;
-import static org.greenplum.pxf.api.io.DataType.TIMESTAMP;
-import static org.greenplum.pxf.api.io.DataType.UNSUPPORTED_TYPE;
-import static org.greenplum.pxf.api.io.DataType.VARCHAR;
-import static org.greenplum.pxf.api.io.DataType.VARCHARARRAY;
+import static org.greenplum.pxf.api.io.DataType.*;
```
Comment (denalex) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java:23]: we configure IntelliJ to not use `.*` imports

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java [hunk 7]
```diff
@@ -281,6 +281,10 @@ private void ensureFunctionsAreInitialized() {
                     functions[i] = ORCVectorizedMappingFunctions::timestampMapper;
                     typeOidMappings[i] = TIMESTAMP.getOID();
                     break;
+                case TIMESTAMP_INSTANT:
+                    functions[i] = ORCVectorizedMappingFunctions::timestampWithTimeZoneMapper;
+                    typeOidMappings[i] = TIMESTAMP_WITH_TIME_ZONE.getOID();
```
[the remark this answers is not in the document]
Comment (denalex) [reply server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java:263]: there should be no changes to this class if the branch is based on `write-orc-tests`

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java [hunk 8]
```diff
@@ -112,20 +112,37 @@ public void testTimestampPushDown() throws Exception {
...
+        assertRowsReturned(NO_ROWS);
+    }
+
     @Test
     public void testRealPushDown() throws Exception {
         // ORC SearchArgument filters out row groups. Since we have a single
         // group, all rows will be returned
         // r > 7 and r < 8
-        context.setFilterString("a6c701s1d7o2a6c700s1d8o1l0");
+        context.setFilterString("a7c701s1d7o2a7c700s1d8o1l0");
```
Comment (denalex) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:140]: can we just add the new type as the last one in the test schema and avoid all these changes ?

File: server/pxf-hdfs/src/test/resources/orc/Makefile [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# default assumes ORC tools JAR is downloaded in current directory
+# download from https://search.maven.org/artifact/org.apache.orc/orc-tools/1.7.5/jar
+ORC_TOOLS_JAR ?= orc-tools-1.7.5-uber.jar
```
Comment (denalex) [inline server/pxf-hdfs/src/test/resources/orc/Makefile:3]: on Mac I use `orc-tools` command that was brew installed and is a wrapper. Its version is 1.7.4 I wonder if we can / should standardize on using brew-installed ORC tools on Mac. Just a thought, not a requirement.

File: server/pxf-hdfs/src/test/resources/orc/README.md [hunk 10]
```diff
@@ -6,109 +6,56 @@ follow the instructions below.
...
-./generate_orc_types.bash
-```
-Similar to the files listed above, the `orc_types_compound_multi.orc` file is generated through insert statements in the HQL.
-If desired, you can copy down the CSV file by running the following command:
-```
-mv "./orc_types_compound_multi/000000_0" "./${CSV_FILENAME}"
+make orc_types_repeated.orc
 ```
 
+
```
Comment (denalex) [inline server/pxf-hdfs/src/test/resources/orc/README.md:58]: I think it is a good idea to generate ORC files from JSON instead of requiring to use Hive. However, I wonder why this was not done in the first place and if we will lose anything Hive-specific ?

Comment (bradfordb-vmware) [reply server/pxf-hdfs/src/test/resources/orc/README.md:58]: In older versions of orc-tools (1.6.x), `date` was not a supported type so it wasn't possible to use orc-tools to convert directly from CSV/JSON to ORC files. Now that 1.7.x is available, `date`, `timestamp`, etc. are supported.

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 11]
```diff
@@ -347,10 +350,15 @@ private void assertDataReturned(Object[][] expected, List<List<OneField>> fields
                     Object expectedValue = expected[colNum][rowNum];
                     if (colNum == 4 && expectedValue != null) {
                         expectedValue = new HiveDecimalWritable(String.valueOf(expectedValue));
-                    } else if (colNum == 11 && expectedValue != null) {
+                    } else if (colNum == 12 && expectedValue != null) {
                         expectedValue = Date.valueOf(String.valueOf(expectedValue));
+                    }else if (colNum==6 && expectedValue!=null){
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:355]:
```suggestion
                    } else if (colNum == 6 && expectedValue != null){
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 12]
```diff
@@ -347,10 +350,15 @@ private void assertDataReturned(Object[][] expected, List<List<OneField>> fields
...
+                        DateTimeFormatter dateTimeFormatter= DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssX");
+                        expectedValue= ZonedDateTime.parse(String.valueOf(expectedValue),dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:357]:
```suggestion
                        expectedValue = ZonedDateTime.parse(String.valueOf(expectedValue),dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 13]
```diff
@@ -347,10 +350,15 @@ private void assertDataReturned(Object[][] expected, List<List<OneField>> fields
...
+                        dateTimeFormatter= DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssXXX");
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:358]:
```suggestion
                        dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssXXX");
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 14]
```diff
@@ -347,10 +350,15 @@ private void assertDataReturned(Object[][] expected, List<List<OneField>> fields
...
+                        value=ZonedDateTime.parse(String.valueOf(value), dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:359]:
```suggestion
                        value = ZonedDateTime.parse(String.valueOf(value), dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 15]
```diff
@@ -411,23 +419,31 @@ private void assertSubsetOfDataReturned(List<List<OneField>> fieldsForBatch) {
                         case 5:
                             expectedValue = COL6_SUBSET[rowNum];
                             break;
-                        case 8:
-                            expectedValue = COL9_SUBSET[rowNum];
+                        case 6:
+                            if(COL7_SUBSET[rowNum] != null){
+                                DateTimeFormatter dateTimeFormatter= DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssX");
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:424]:
```suggestion
                                DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssX");
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 16]
```diff
@@ -411,23 +419,31 @@ private void assertSubsetOfDataReturned(List<List<OneField>> fieldsForBatch) {
...
+                                expectedValue= ZonedDateTime.parse(String.valueOf(COL7_SUBSET[rowNum]),dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:425]:
```suggestion
                                expectedValue = ZonedDateTime.parse(String.valueOf(COL7_SUBSET[rowNum]),dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 17]
```diff
@@ -411,23 +419,31 @@ private void assertSubsetOfDataReturned(List<List<OneField>> fieldsForBatch) {
...
+                                dateTimeFormatter= DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssXXX");
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:426]:
```suggestion
                                dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ssXXX");
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java [hunk 18]
```diff
@@ -411,23 +419,31 @@ private void assertSubsetOfDataReturned(List<List<OneField>> fieldsForBatch) {
...
+                                value=ZonedDateTime.parse(String.valueOf(value), dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverTest.java:427]:
```suggestion
                                value = ZonedDateTime.parse(String.valueOf(value), dateTimeFormatter).withZoneSameInstant(ZoneOffset.UTC);
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedAccessorTest.java [hunk 19]
```diff
@@ -94,12 +94,13 @@ public void testReadSubsetOfOrcTypesFile() throws IOException {
         columnDescriptors.add(new ColumnDescriptor("t1", DataType.TEXT.getOID(), 0, "text", null));
         columnDescriptors.add(new ColumnDescriptor("num1", DataType.INTEGER.getOID(), 2, "int4", null));
         columnDescriptors.add(new ColumnDescriptor("tm", DataType.TIMESTAMP.getOID(), 5, "timestamp", null));
-        columnDescriptors.add(new ColumnDescriptor("r", DataType.REAL.getOID(), 6, "real", null));
-        columnDescriptors.add(new ColumnDescriptor("tn", DataType.SMALLINT.getOID(), 9, "int2", null));
-        columnDescriptors.add(new ColumnDescriptor("dt", DataType.DATE.getOID(), 11, "date", null));
-        columnDescriptors.add(new ColumnDescriptor("bin", DataType.BYTEA.getOID(), 14, "bin", null));
+        columnDescriptors.add(new ColumnDescriptor("tmtz", DataType.TIMESTAMP_WITH_TIME_ZONE.getOID(),6,"timestamp with time zone",null));
```
Comment (hpandeycodeit) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedAccessorTest.java:97]:
```suggestion
        columnDescriptors.add(new ColumnDescriptor("tmtz", DataType.TIMESTAMP_WITH_TIME_ZONE.getOID(), 6, "timestamp with time zone", null));
```

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java [hunk 20]
```diff
@@ -113,7 +113,6 @@ public class ORCVectorizedResolver extends BasePlugin implements ReadVectorizedR
      * The array has the same size as the writeSchema, and the functions depend on the type of the elements in the schema.
      */
     private TriConsumer<ColumnVector, Integer, Object>[] writeFunctions;
-
```
Comment (ashuka24) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java:116]: nit: for consistency, let's keep this spacing

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java [hunk 21]
```diff
@@ -31,8 +31,8 @@
 import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
 import java.time.ZoneId;
-import java.time.ZonedDateTime;
```
Comment (ashuka24) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java:34]: nit: not sure why this change was made.

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java [hunk 22]
```diff
@@ -138,8 +137,6 @@ public class ORCVectorizedResolver extends BasePlugin implements ReadVectorizedR
     private List<List<OneField>> cachedBatch;
     private VectorizedRowBatch vectorizedRowBatch;
 
-    private static final String UNSUPPORTED_ERR_MESSAGE = "Current operation is not supported";
-
```
Comment (ashuka24) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolver.java:142]: nit: same here. let's keep the spacing for consistency

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java [hunk 23]
```diff
@@ -294,32 +295,32 @@ private void assertTypes(List<OneField> fieldList) {
         if (columnDescriptors.get(5).isProjected()) {
             assertEquals(DataType.TIMESTAMP.getOID(), fieldList.get(5).type);
         }
-        if (columnDescriptors.get(6).isProjected()) {
```
Comment (ashuka24) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:297]: do we just skip over checking this column?

Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:297]: Can we add a comment here about skipping column index 6 because we're not supporting predicate pushdown on that column type?

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java [hunk 24]
```diff
@@ -393,15 +388,15 @@ private void assertValues(List<OneField> fieldList, final int row) {
         }
 
         if (columnDescriptors.get(11).isProjected() && COL12[row] != null) {
-            assertEquals(Date.valueOf(COL12[row]), fieldList.get(11).val, "Row " + row);
+            assertEquals((COL12[row]), fieldList.get(11).val, "Row " + row);
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:391]:
```suggestion
            assertEquals(COL12[row], fieldList.get(11).val, "Row " + row);
```

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java [hunk 25]
```diff
@@ -362,12 +363,6 @@ private void assertValues(List<OneField> fieldList, final int row) {
             assertNull(fieldList.get(5).val, "Row " + row);
         }
 
-        if (columnDescriptors.get(6).isProjected() && COL7[row] != null) {
```
Comment (ashuka24) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:365]: same here, why are we no longer checking the value in COL7?

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverReadTest.java [hunk 26]
```diff
@@ -347,10 +353,13 @@ private void assertDataReturned(Object[][] expected, List<List<OneField>> fields
                     Object expectedValue = expected[colNum][rowNum];
                     if (colNum == 4 && expectedValue != null) {
                         expectedValue = new HiveDecimalWritable(String.valueOf(expectedValue));
-                    } else if (colNum == 11 && expectedValue != null) {
+                    } else if (colNum == 12 && expectedValue != null) {
                         expectedValue = Date.valueOf(String.valueOf(expectedValue));
+                    } else if (colNum == 6 && expectedValue != null) {
```
Comment (ashuka24) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedResolverReadTest.java:358]: nit: can we order these so that the if check for col 6 comes before the if check for col 12

File: server/pxf-hdfs/src/test/resources/orc/README.md [hunk 27]
```diff
@@ -64,51 +40,25 @@ where you ran the script.
...
```
Comment (ashuka24) [inline server/pxf-hdfs/src/test/resources/orc/README.md:61]: nit: extra new line

File: server/pxf-hdfs/src/test/resources/orc/README.md [hunk 28]
```diff
@@ -6,53 +6,29 @@ follow the instructions below.
...
 
-Note: The scripts will run with a older version of Hive. However, to get the
-correct schema inside the ORC files Hive 2.3+ is required. Please double check
-that the Hive version is correct.
-
-## Generate the orc_types.orc file
-
-Identify your hdfs and hive commands. Identify the Hive warehouse path where
-table data gets stored, for example:
+Note: If you download orc-tools-1.7.5-uber.jar not into the current project path, you will need to:
```
Comment (bradfordb-vmware) [inline server/pxf-hdfs/src/test/resources/orc/README.md:12]:
```suggestion
Note: If you don't download orc-tools-1.7.5-uber.jar into the current project path, you will need to:
```

**Review by bradfordb-vmware — APPROVED** [review]

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java (hunk 23 above)
Comment (denalex) [reply server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:297]: I am not sure this only asserts types that are supported for pushdown, it seems it asserts types of the returned columns, so I think we should be asserting the new type even if we do not support pushdown

File: server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java [hunk 29]
```diff
@@ -393,15 +388,15 @@ private void assertValues(List<OneField> fieldList, final int row) {
...
         } else {
             assertNull(fieldList.get(11).val);
         }
 
         if (columnDescriptors.get(12).isProjected() && COL13[row] != null) {
-            assertEquals(COL13[row], fieldList.get(12).val, "Row " + row);
+            assertEquals(Date.valueOf(COL13[row]), fieldList.get(12).val, "Row " + row);
         } else {
-            assertNull(fieldList.get(12).val, "Row " + row);
+            assertNull(fieldList.get(12).val);
```
Comment (denalex) [inline server/pxf-hdfs/src/test/java/org/greenplum/pxf/plugins/hdfs/orc/ORCFilterPushDownTest.java:399]: wonder why `"Row " + row` was removed, it seems it is used everywhere else

**Review by denalex — APPROVED** [review]
